Gems/Atom/Feature/Common/Code/Source/ReflectionScreenSpace/ReflectionScreenSpaceDownsampleDepthLinearPass.cpp

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "ReflectionScreenSpaceDownsampleDepthLinearPass.h"
+#include "ReflectionScreenSpaceDownsampleDepthLinearChildPass.h"
+#include <Atom/RPI.Public/RPIUtils.h>
+
+namespace AZ
+{
+    namespace Render
+    {
+        RPI::Ptr<ReflectionScreenSpaceDownsampleDepthLinearPass> ReflectionScreenSpaceDownsampleDepthLinearPass::Create(const RPI::PassDescriptor& descriptor)
+        {
+            RPI::Ptr<ReflectionScreenSpaceDownsampleDepthLinearPass> pass = aznew ReflectionScreenSpaceDownsampleDepthLinearPass(descriptor);
+            return AZStd::move(pass);
+        }
+
+        ReflectionScreenSpaceDownsampleDepthLinearPass::ReflectionScreenSpaceDownsampleDepthLinearPass(const RPI::PassDescriptor& descriptor)
+            : RPI::ParentPass(descriptor)
+        {
+        }
+
+        void ReflectionScreenSpaceDownsampleDepthLinearPass::ResetInternal()
+        {
+            RemoveChildren();
+        }
+
+        void ReflectionScreenSpaceDownsampleDepthLinearPass::CreateChildPassesInternal()
+        {
+            RPI::PassSystemInterface* passSystem = RPI::PassSystemInterface::Get();
+
+            // load shader
+            AZStd::string shaderFilePath = "Shaders/Reflections/ReflectionScreenSpaceDownsampleDepthLinear.azshader";
```

Comment:
hmm do we need to add the shader to some engine seed list somewhere for release builds?